From 1adcbdc137e0ad0f8b0e6a7c4d35fdcae83e5778 Mon Sep 17 00:00:00 2001
From: Todd Whiteman <todd.whiteman@joyent.com>
Date: Tue, 7 May 2019 09:16:41 -0700
Subject: [PATCH] TRITON-1623 vmapi migration test has intermittent 'bandwidth
 progress event was seen' failure Reviewed by: Orlando Vazquez
 <orlando@joyent.com> Approved by: Orlando Vazquez <orlando@joyent.com>

---
 lib/vm-migration/migrate.js | 6 ++++++
 package.json                | 2 +-
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/lib/vm-migration/migrate.js b/lib/vm-migration/migrate.js
index db92d27..f1209de 100644
--- a/lib/vm-migration/migrate.js
+++ b/lib/vm-migration/migrate.js
@@ -801,6 +801,12 @@ function watchVmMigration(req, res, next) {
         // running).
         if (!MIGRATION_RECORD_FOR_VM_UUID[vmUuid] ||
                 MIGRATION_RECORD_FOR_VM_UUID[vmUuid].state !== 'running') {
+            // Send out any pending progress events first.
+            if (MIGRATION_PROGRESS_FOR_VM_UUID[vmUuid]) {
+                res.write(JSON.stringify(MIGRATION_PROGRESS_FOR_VM_UUID[vmUuid])
+                    + '\n');
+            }
+            // Send the end event.
             res.write(JSON.stringify(
                 {
                     type: 'end',
diff --git a/package.json b/package.json
index c1c484d..b6fde8d 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "vmapi",
   "description": "VMs API",
-  "version": "9.8.5",
+  "version": "9.8.6",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
-- 
2.21.0

